test/client.js

```diff
@@ -591,8 +632,8 @@ describe('Rest Client', function () {
       var statusCode = 418;
       api.post(path, requestBody).reply(statusCode);
       client.post(url, requestBody, function (err, body, resp) {
-          assert.equal(resp.statusCode, statusCode);
-          done();
+        assert.equal(resp.statusCode, statusCode);
```

Comment:
Can add a assert(err). We don't want an assert.ifError() as it's supposed to be an error in this case (418)